mockito-core/src/main/java/org/mockito/internal/PremainAttachAccess.java

```diff
@@ -34,16 +40,41 @@ public static Instrumentation getInstrumentation() {
         }
         if (instrumentation == null) {
             if (ClassFileVersion.ofThisVm().isAtLeast(ClassFileVersion.JAVA_V21)) {
-                // Cannot use `Plugins.getMockitoLogger().warn(...)` at this time due to a circular
-                // dependency on `Plugins.registry`.
-                // The `PluginRegistry` is not yet fully initialized (in `Plugins`), because it is
-                // currently initializing the `MockMaker` which is a InlineByteBuddyMockMaker, and
-                // it is later calling this method to access the instrumentation.
-                System.err.println(
-                        "Mockito is currently self-attaching to enable the inline-mock-maker. This "
-                                + "will no longer work in future releases of the JDK. Please add Mockito as an agent to your "
-                                + "build as described in Mockito's documentation: "
-                                + "https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#0.3");
+                boolean dynamicAgentLoading;
+                try {
+                    Object runtimeMXBean =
+                            Class.forName("java.lang.management.ManagementFactory")
```

Comment:
Why do we need to use reflection here?

Reply:
Android does not know these classes and would fail if Mockito had a hard link to them. Now the invocation will just fail.